DasharoBootPolicies/BootPolicies.c

```diff
@@ -143,5 +145,23 @@ InitializeBootPolicies (
     DEBUG ((EFI_D_INFO, "Boot Policy: Enabling PS2 Controller\n"));
   }
 
+  Status = GetVariable2 (
+           L"IommuHandoff",
+           &gDasharoSystemFeaturesGuid,
+           (VOID **) &EfiVar,
+           &VarSize
+           );
+
+  if ((Status != EFI_NOT_FOUND) && (VarSize == sizeof(*EfiVar))){
```

Comment:
I think it should be

```
if (Status != EFI_NOT_FOUND) {
...
} else if ((Status == EFI_SUCCESS) && (VarSize == sizeof(*EfiVar))) {
...
}
```
Because there is no guarantee that size will be returned correctly. I found out this issue when networkbootdefautlenable was not working as expected

Reply:
I'm not sure I quite understand, shouldn't it be like:
```
if (Status == EFI_NOT_FOUND) {
...
} else if ((Status == EFI_SUCCESS) && (VarSize == sizeof(*EfiVar))) {
...
}

```
?
From what I see the `else if` will never be executed. And if yes, what error handling do we want when we get the wrong size, do we just throw an error?

Reply:
@Stojek139808 I think your idea is correct, probably @miczyg1 did a typo. I think that wrong size should behave the same way as `EFI_NOT_FOUND` does, i.e. fall back to default value, after printing a debug output. If someone manually messed up the variable contents, this is the only reasonable way of fixing it.

Reply:
Pushed a change.